### PR TITLE
Manage job offers

### LINKED ISSUE #6: Company can manage job offers
Once a company gets their account verified(#7), they can post a job offer.
The job offer must be approved by an admin.

### PR DESCRIPTION
**Pull request by ianrodrigues**
Closes #6

### CODE REVIEW

File: src/Controller/Api/JobOfferController.php [hunk 1]
```diff
@@ -62,4 +66,28 @@ public function display(int $jobOfferId): Response
             AbstractNormalizer::GROUPS => ['detail']
         ]);
     }
+
+    /**
+     * @IsGranted("ROLE_COMPANY")
+     * @Route("/api/job-offer", name="api_job_offer_create", methods={"POST"})
```
Comment (rtio) [inline src/Controller/Api/JobOfferController.php]: This route should be named /job-offers/ as a POST method

File: src/Entity/JobOffer.php [hunk 2]
```diff
@@ -82,6 +84,18 @@ public function __toString(): string
         return "#{$this->id} {$this->title}";
     }
 
+    public static function createFromPost(PostJobOffer $data, Company $company): self
+    {
+        return (new static)
```
Comment (rtio) [inline src/Entity/JobOffer.php]: (new static())

File: src/Service/JobOfferService.php [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+namespace App\Service;
+
+use App\Entity\Company;
+use App\Entity\JobOffer;
+use App\Request\PostJobOffer;
+use App\Request\CompanyRegistration;
+use App\Repository\JobOfferRepository;
+
+final class JobOfferService
```
Comment (rtio) [inline src/Service/JobOfferService.php]: Should we add a test for this service?

File: src/Controller/Api/JobOfferController.php (hunk 1 above)
Comment (rtio) [reply src/Controller/Api/JobOfferController.php]: There is a lot of discussion about this one. Here are some texts about it https://www.moesif.com/blog/api-guide/api-design-guidelines/#plural-vs-singular-resource-names. Btw I think we need to create something like a guide to keep us on the same page. Thoughts?

Comment (ianrodrigues) [reply src/Controller/Api/JobOfferController.php]: Plural is okay to me, actually it was a typo.

File: src/Service/JobOfferService.php (hunk 3 above)
Comment (ianrodrigues) [reply src/Service/JobOfferService.php]: I don't think so. We already have it covered by integration test. I can't see any edge case that'd deserve a unit test.

File: src/Entity/JobOffer.php (hunk 2 above)
Comment (ianrodrigues) [reply src/Entity/JobOffer.php]: thanks!
